### ISSUE TITLE
private/public methods are not recognised

### ISSUE THREAD

**Issue by arcanist123** [abaplint/abaplint#1990]
It should not be possible to call the private methods of the class. this report should produce a syntax error  
```abap
REPORT z_test_transpiler_2.


CLASS demo DEFINITION.
  PUBLIC SECTION.
    CLASS-METHODS:
      main.
ENDCLASS.




CLASS  lcl_ DEFINITION CREATE PRIVATE.

  PUBLIC SECTION.

  PRIVATE SECTION.
    CLASS-METHODS:
      private_method.
ENDCLASS.

CLASS lcl_ IMPLEMENTATION.
  METHOD private_method.
    WRITE: 'something'.
  ENDMETHOD.

ENDCLASS.

CLASS demo IMPLEMENTATION.
  METHOD main.
    lcl_=>private_method( ).
  ENDMETHOD.


ENDCLASS.

START-OF-SELECTION.
  demo=>main(  ).
```
